rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt

```diff
@@ -71,20 +73,62 @@ public sealed class BaseChoiceBuilder<T>(
     )
 }
 
+/**
+ * Builder for numeric options.
+ */
+public sealed class NumericOptionBuilder<T : Number>(
+    name: String,
+    description: String,
+    type: ApplicationCommandOptionType
+) : BaseChoiceBuilder<T>(name, description, type) {
+
+    private var _minValue: Optional<T> = Optional.Missing()
+
+    /**
+     * The minimum value permitted.
+     */
+    public var minValue: T? by ::_minValue.delegate()
+
+    private var _maxValue: Optional<T> = Optional.Missing()
+
+    /**
+     * The maximum value permitted.
+     */
+    public var maxValue: T? by ::_maxValue.delegate()
+
+    override fun toRequest(): ApplicationCommandOption = ApplicationCommandOption(
+        type,
+        name,
+        description,
+        choices = _choices,
+        required = _required,
+        default = _default,
+        autocomplete = _autocomplete,
+        minValue = _minValue.map { JsonPrimitive(it) },
+        maxValue = _maxValue.map { JsonPrimitive(it) },
+    )
+}
+
+
+@Deprecated("Replaced by IntOptionBuilder", ReplaceWith("IntOptionBuilder"))
+public typealias IntChoiceBuilder = IntOptionBuilder
+
 @KordDsl
-public class IntChoiceBuilder(name: String, description: String) :
-    BaseChoiceBuilder<Long>(name, description, ApplicationCommandOptionType.Integer) {
+public class IntOptionBuilder(name: String, description: String) :
+    NumericOptionBuilder<Long>(name, description, ApplicationCommandOptionType.Integer) {
```

Comment:
If you already rename this, I would suggest `IntegerOptionBuilder`, it makes the distinction from `Int` clearer, we already have `ApplicationCommandOptionType.Integer`, `IntegerArgument` and `IntegerOptionValue`.